### ISSUE TITLE
0xc000007b error when running c++ examples in visual studio 

### ISSUE THREAD

**Issue by MustafaYounes1** [microsoft/onnxruntime#19976]
### Describe the issue

Whenever I try to run the c++ examples using the Visual Studio 2019 Local Windows Debugger, I get `0xc000007b` error.

After scrolling a bit, I found that this could be an indicator of missed `.dll`s, so I tried to add the path to the installed onnxruntime `bin` folder to my  Environment via the `<LocalDebuggerEnvironment>` property  in the `vs.props` config file.

Notes: 
- The built executables have the required `.dll`s beside them, and I can run them successfully.
- onnxruntime was built as a shared library.

### To reproduce

With VS 2019:
- Build onnxruntime 1.15.1 from source
- Build the c++ inference examples
- run any application with the Local Windows Debugger

### Platform

Windows

### OS Version

10

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

1.15.1

### ONNX Runtime API

C++

### Architecture

X64

### Execution Provider

Default CPU

**Comment by mszhanyi** [comment]
Could I know why are you still building onnxruntime 1.15.1?
Since onnxruntime 1.16, We're using VS2022 to build the source.

And Can you provide more details?
1. Which inference example are you using. Is it in https://github.com/microsoft/onnxruntime-inference-examples?
2. I think you don't need to build the souce, you could download the zipped binaries directly https://github.com/microsoft/onnxruntime/releases/tag/v1.17.1, 
for example: https://github.com/microsoft/onnxruntime/releases/download/v1.17.1/onnxruntime-training-win-x86-1.17.1.zip
4. You could checkout the CMakeLists.txt in https://github.com/microsoft/onnxruntime-inference-examples/tree/main/c_cxx/customop to build a simple onnxruntime program.

**Comment by MustafaYounes1** [comment]
@mszhanyi thanks for the quick reply.

- First of all, unfortunately, I'm restricted to VS2019. And according to the [announcements](https://github.com/microsoft/onnxruntime/releases/tag/v1.15.0) of `v1.15.0`, starting from `v1.16.0`, the support would be dropped for VS2019
- I have to build from source as well. I'm working on multi-platform project, and using prebuilt binaries is not a viable option for me. (VS2019 is the IDE I work on though)
- all [c_xx examples](https://github.com/microsoft/onnxruntime-inference-examples/tree/main/c_cxx) failed when executing them in the VS2019 Local Windows Debugger.

What's noteworthy is that, when I build the examples, everything goes smoothly, but when it comes to running the executables using the VS Local Windows Debugger, the `0xc000007b` crash occurs.

[6243](https://github.com/microsoft/onnxruntime/issues/6243) is a similar issue. I have the `dll`s beside my executables, and I can run them using a command prompt, but the VS Local Windows Debugger crashes with `0xc000007b` even though I added the onnxruntime bin path to my Environment as I described earlier.

**Comment by mszhanyi** [comment]
It looks most likely an environment issue of visual studio.
If you just want to use VS to debug your code.
One workaround is to add DebugBreak() in your executable source code.
Then you could run your executable in command, it would give you an opportunity to use VS to attatch to the process.

**Comment by MustafaYounes1** [comment]
I don't think it's an env issue of VS2019, cause I provided the env will all required paths to dlls.

It turned out that it was the same issue as [6243](https://github.com/microsoft/onnxruntime/issues/6243). 

Even though I extended my VS `%PATH%` with the paths to the `onnxruntime x64 dlls` I built, It looks for `dlls` in your system first, unless it finds the required `dlls` beside the built binaries. 

I used cmake as the build system generator, and I put the dlls where I "installed" the binaries of my c++ app but forgot to put them beside the target's `RUNTIME_OUTPUT_DIRECTORY`. 

And that's why the VS Local Windows Debugger failed to load the `x64 dlls` I built and tried to load `x86 dlls` found in the `C:\Windows\System32`, hence the `0xc000007b` crash.

Required `dlls` could be automatically copied to where your runtime targets would be built with the cmake function defined [here](https://github.com/microsoft/onnxruntime-inference-examples/blob/main/c_cxx/CMakeLists.txt#L81)
